### ISSUE TITLE
<pre> tag html entities

### ISSUE THREAD

**Issue by redrails** [anchorcms/anchor-cms#1126]
### Summary

I can't see that this is already reported but I have been using AnchorCMS for my codeblog and writing code like:

    List<String> stringList = new ArrayList<>();

And wrapping it around `<pre>` tags removes the pointy-brackets due to stripping already applied before `article_markdown();`. This is inconvenient and is causing loss of code when viewing the blog.

### Expected Behaviour

- Should be displayed without escaping all htmlentities.

### Actual Behaviour

```
<pre>
    List<String> stringList = new ArrayList<>();
</pre>
```
outputs:

    List stringList = new ArrayList(); // note < and > and everything between those tags is removed

### My fix
In `article.php` I have modified the `article_markdown()` function to be the following:

	function article_markdown() {
		$content = Registry::prop('article', 'html');

		$string_new = preg_replace_callback(

		    '#<pre>([\\s\\S]+?)</pre>#',
		    create_function(
		        '$input',
		        'return "<pre>".htmlentities($input[1])."</pre>";'
		    ),
		    $content
		);

		return $string_new;
	}

Credit to [bone-idle](http://stackoverflow.com/questions/17897695/check-string-and-htmlentities-everything-within-the-pre/17897891#17897891) for the preg_replace code.

### Context details (if applicable)

- Anchor version: 0.12.1
- Server setup: Apache 2.2.15 / PHP 5.5.32

**Comment by daviddarnes** [comment]
Are you not using the back tick method for code fencing? See https://help.github.com/articles/creating-and-highlighting-code-blocks/

**Comment by redrails** [comment]
I think I overlooked full markdown support possibly, I tried doing the back tick code block method previously with another theme but it failed to work so I didn't bother with it again. However it seems to be working now but I think if someone is not using markdown, the above REGEX replacement method may still be useful. Sorry for the hassle.

**Comment by daviddarnes** [comment]
@redrails no problem, thank you for reporting and suggesting a solution 👍

**Comment by TheBrenny** [comment]
@daviddarnes, can this have a discussion tag added to it? I think it might be worthwhile to have more input on this (even though it's closed). Should we still implement this RegEx? Should we convert and `<pre>`s and `</pre>`s to their markdown equivalent (<code>```</code>)?

###### (Slightly ironic how you need to fence the backticks in `<code>` blocks on Github. :joy_cat:)

**Comment by redrails** [comment]
@TheBrenny, actually the reason I resorted to modifying and using `<pre>` was due to the styling I wanted to apply to both the `<pre>` and `<code>` tags. So I presumed that if I use the back-tick md method it may not like the custom CSS. I would've thought that by default `<pre>` is handled for `htmlentities()` and not `htmlspecialchars()` as that would cause some string literals to not be handled correctly when writing code.
